lib/std/testing.zig

```diff
@@ -299,6 +299,41 @@ pub fn expectEqualSlices(comptime T: type, expected: []const T, actual: []const
     }
 }
 
+/// This function is intended to be used only in tests. Checks that two slices or two arrays are equal,
+/// including that their sentinel (if any) are the same. Will error if given another type.
+pub fn expectEqualSlicesSentinel(expected: anytype, actual: anytype) !void {
```

Comment:
How about making this more generic, just `expectEqualSlices`, and you check the sentinel if the slice type indicates there is a sentinel?

Reply:
In fact, you could just extend `expectEqual` to do all this as well.

Reply:
We talked on Discord, and tried to come up with a way to simply change `expectEqualSlices` but didn't find something simple enough that was worth making.
I also tried simply calling `expectEqualSlices` instead of re-writing it, but it turned into a bit of a mess to make sure that each argument (which can be a pointer or an array) was properly cast-ed to pass to `expectEqualSlices`. That turned out to be more complex code than just doing the same iteration as `expectEqualSlices`, so I ended up going with that method.